app/upgrades/v7/upgrades.go

```diff
@@ -0,0 +1,199 @@
+			return vm, errorsmod.Wrapf(err, "unable to modify juno unbonding frequency")
+		}
+
+		// Incentive diversification
+		if err := IncentiveDiversification(ctx, bankKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to send ustrd tokens for prop #153 (incentive diversification)")
+		}
+
+		// Create reward collector module account
+		if err := CreateRewardCollectorModuleAccount(ctx, accountKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable create reward collector module account")
+		}
+
+		ctx.Logger().Info("Running module mogrations...")
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// Add a new hourly epoch that will be used by the rate limit module
+func AddHourEpoch(ctx sdk.Context, k epochskeeper.Keeper) {
+	ctx.Logger().Info("Adding hour epoch")
+
+	hourEpoch := epochstypes.EpochInfo{
+		Identifier:              epochstypes.HOUR_EPOCH,
+		StartTime:               time.Time{},
+		Duration:                time.Hour,
+		CurrentEpoch:            0,
+		CurrentEpochStartHeight: 0,
+		CurrentEpochStartTime:   time.Time{},
+		EpochCountingStarted:    false,
+	}
+
+	k.SetEpochInfo(ctx, hourEpoch)
+}
+
+// Increase stride inflation from 285 STRD to 1078 STRD
+func IncreaseStrideInflation(ctx sdk.Context, k mintkeeper.Keeper) {
+	ctx.Logger().Info("Increasing STRD inflation")
+
+	epochProvisions := sdk.NewDec(1_078_767_123)
+	minter := minttypes.NewMinter(epochProvisions)
+	k.SetMinter(ctx, minter)
+}
+
+// Add stride messages (LiquidStake, RedeemStake, Claim) to the ICAHost allow messages
+// to allow other protocols to liquid stake via ICA
+func AddICAHostAllowMessages(ctx sdk.Context, k icahostkeeper.Keeper) {
+	ctx.Logger().Info("Adding ICA Host allow messages")
+
+	params := icahosttypes.Params{
+		HostEnabled: true,
+		AllowMessages: []string{
+			sdk.MsgTypeURL(&banktypes.MsgSend{}),
+			sdk.MsgTypeURL(&banktypes.MsgMultiSend{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgUndelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
+			sdk.MsgTypeURL(&distrtypes.MsgWithdrawDelegatorReward{}),
+			sdk.MsgTypeURL(&distrtypes.MsgSetWithdrawAddress{}),
+			sdk.MsgTypeURL(&transfertypes.MsgTransfer{}),
+			sdk.MsgTypeURL(&govtypes.MsgVote{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgLiquidStake{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgRedeemStake{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgClaimUndelegatedTokens{}),
+		},
+	}
+	k.SetParams(ctx, params)
+}
+
+// Add the min/max redemption rate to each host zone as safety bounds, using the default for each
+// Also set the Halted boolean to false
+func AddRedemptionRateSafetyChecks(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	ctx.Logger().Info("Setting min/max redemption rate safety bounds on each host zone")
+
+	// Get default min/max redemption rate
+	defaultMinRedemptionRate := sdk.NewDecWithPrec(int64(k.GetParam(ctx, types.KeyDefaultMinRedemptionRateThreshold)), 2)
```

Comment:
I think we need to set this param before getting it - should this happen before running `AddRedemptionRateSafetyChecks`?

Reply:
Done

Reply:
great catch! Thanks for resolving